### ISSUE TITLE
update ssh button on IDE for Openshift

### ISSUE THREAD

**Issue by gorkem** [eclipse-che/che#7664]
The SSH button on the IDE provides information about how to ssh into a workspace. We need to update this information so that when a workspace is running on Openshift it provides information for `oc rsh` based information. For other infrastructures the information should also check the existence of ssh agent and provide information for enabling the agent if it is not already enabled.

**Comment by skabashnyuk** [comment]
There is a couple of different modes how Che starting workspaces. 
For Che6 now we using a predefined account that can run workspaces on OCP. However other services like OSIO can use Che 6 in multiuser mode, where workspace started under the user account. But it requires some additional services to identify a user (Keycloak or Auth service) and namespace where we can start workspaces. 

That means that for now, Che6 on OCP ```oc rsh``` is not a good choice because it'll require providing predefined user credentials. The only choice left is when sudo is allowed on OCP cluster so we can run sshd demon. From another hand, we may plan to adapt user specific flow(when workspace started in user namespace) on pure OCP but we have to clarify a couple of moment before we can start implementing it.
1. How the user will be authenticated. Does OCP provide it own keycloak? Or how OCP can use Che provided keycloak
2. How will we know user namespace where to run Che workspace?

**Comment by skabashnyuk** [comment]
In this sprint planned two task https://github.com/eclipse/che/issues/7624
https://github.com/eclipse/che/issues/7908 after their implementation by default there will be not ssh buttons because there is no ssh server defined by default in the workspace.
@eivantsov  is going to add information in the documentation for the case if the user knows credentials under which we are running all workspaces. With this credentials, user will be able to run
```oc rsh``` to have pseudo tty to execute some commands. Also, I've created an epic to use openshift account to create workspace objects https://github.com/eclipse/che/issues/8178. 
In this case, we can add some panel in case if we are running on openshift in this mode to tell individual users that they may use ```oc rsh``` on behalf of their names to have pseudo tty in his workspace.

@eivantsov  please comment when you'll add some notes to docs about how user is able to use ```oc rsh``` now so after that @gorkem @slemeur can validate is this acceptable solution for now and Che 6 GA or we need some future actions.

**Comment by gorkem** [comment]
We do not need to over do it. Here is a template that should work for all OS flavors. We just need to inject the `{$url for openshift}` and the `{$podname}` which should not be an issue. Injecting the actual token is a bit of a security concern anyway.

```
You can connect to your workspace via OpenShift Container Platform CLI :

[ALL OS]
 oc login {$url to openshift} --token=<API token>
 oc rsh {$podname}

For information on the OpenShift Container Platform CLI refer to
https://docs.openshift.com/container-platform/3.7/cli_reference/index.html 
```

**Comment by ghost** [comment]
Is sharing token ok? For all users. Currently, all ws pods are created on behalf of one user whose token or credentials are provided in server configuration.

**Comment by gazarenkov** [comment]
For "one user" mode `oc` is not appropriate, as I understand @skabashnyuk it will be possible after #8178

**Comment by ghost** [comment]
As long as you know credentials used in server config, you can rsh

**Comment by gorkem** [comment]
I do not think giving keys to the one user is a good idea. It probably exposes more privileges than the OS admin intends for regular users.

What are the privileges that we require for that user? Are they documented somewhere?

**Comment by skabashnyuk** [comment]
>What are the privileges that we require for that user?

@gorkem can you rephrase this question?

**Comment by ghost** [comment]
@gorkem no privileges. Currently, Che server either uses credentials/token of one user - basically when you deploy Che to OCP you provide them as envs. So, if you have deployed OCP with your credentials, and I created a workspace using your Che installation, there's no way I can rsh into my ws pod, since it will be **your** pod.

**Comment by gorkem** [comment]
@eivantsov Yes, played with multi-user che a bit over the weekend, I had the wrong assumption on how it was implemented. In this current implementation we are practically shutting user out of any OpenShift tool. I think the best approach for the current state of the implementation is not to show any information about SSH if ssh server is not present.

**Comment by skabashnyuk** [comment]
![2018-01-22 12 36 11 3](https://user-images.githubusercontent.com/1614429/35216686-170b2a86-ff71-11e7-979e-21db9767b99f.png)
Now there is no ssh button by default on OCP since there is no ssh-agent running.

@gorkem would you like to make this issue a part of https://github.com/eclipse/che/issues/8178 ?
Do you want something to be changed before 6 GA?

**Comment by gorkem** [comment]
@skabashnyuk I think we can close this one. The goal was to avoid giving misleading information. I do not think we need to clutter the UI and add an SSH button as part of #8178. 

@eivantsov can make sure we have a section on how to ssh into workspaces running on Openshift on the documentation if we do not already have that.

**Comment by ghost** [comment]
https://www.eclipse.org/che/docs/6/che/docs/openshift-config.html#enable-ssh-and-sudo
